Sage/carbon
# UI Automation tags

## Description

* In some cases Enzyme is added to the specs (all specs should now include Enzyme)
* Tags are added to anything interactable, which includes anything focusable, eg inputs, buttons, toggles
* Further to this they are also added to anything which can be populated with dynamic content, such as titles or content areas, but which might not be interactable
* They are not added to anything that exists for layout or style purposes, such as wrappers
* Anythingi that is not at root level and only contains a prop is also not marked up as these are user defined and therefore already marked up
* Sub-components are named on context, so it would be `data-element='open'` rather than `data-element='button'`
* Sub-divided component, such as Form (with FormSummary) have each of their sub divisions marked up as full components, given the fact they could be used separately and to maintain the componentisation
* `DropdownFilterAjax` doesn't need an open test as open is triggered by AJAX (already tested) and the results are in inherited code (already tested)

Some components exist just for style and layout purposes so have not been included in this pass:
* `AppWrapper`
* `InlineInputs`
* `Row`

Also, refactors some specs to use Enzyme mount/unmount instead of calling lifecycle methods directly

Some path bugs are fixed to be relative

There is also very linter related scoutings

## Todo


## QA Steps

* Load the site up in the demo site
* Quick sanity check components
* This work will be used by automation and thoroughly checked via that route
